File: app/src/main/java/com/thelordash2/bmwe9xtool/VehicleConfig.kt
package com.thelordash2.bmwe9xtool

object VehicleConfig {
    const val PLATFORM = "BMW E8x/E9x"
    const val DATEN_FAMILY = "E89"

    @Volatile
    private var expectedVin: String? = null

    fun configureExpectedVin(vin: String?) {
        expectedVin = vin?.trim()?.uppercase()?.takeIf { it.isNotEmpty() }
    }

    fun vinMatches(readVin: String): Boolean =
        expectedVin != null && readVin.trim().uppercase() == expectedVin
}

File: src/BmwE9xTool.Android/Vehicle/VehicleConfig.cs
namespace BmwE9xTool.Vehicle;

public static class VehicleConfig
{
    public const string Series = "E90";
    public const string NcsChassis = "E89";

    // Expected VIN is deliberately runtime-only. Never commit a vehicle identifier.
    public static string? ExpectedVin { get; private set; }

    public static void ConfigureExpectedVin(string? vin)
    {
        ExpectedVin = string.IsNullOrWhiteSpace(vin) ? null : vin.Trim().ToUpperInvariant();
    }

    public static bool VinMatches(string? vin) =>
        !string.IsNullOrEmpty(ExpectedVin) &&
        string.Equals(vin?.Trim(), ExpectedVin, StringComparison.OrdinalIgnoreCase);
}
